src/test/java/seedu/address/logic/commands/CheckCraftCommandTest.java

```diff
@@ -0,0 +1,171 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CheckCraftCommand.MESSAGE_SUCCESS_CRAFTABLE;
+import static seedu.address.logic.commands.CheckCraftCommand.MESSAGE_SUCCESS_UNCRAFTABLE;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalItems.APPLE;
+import static seedu.address.testutil.TypicalItems.BANANA;
+import static seedu.address.testutil.TypicalItems.getTypicalItemList;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.FilteredList;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.ItemList;
+import seedu.address.model.ReadOnlyItemList;
+import seedu.address.model.ReadOnlyRecipeList;
+import seedu.address.model.RecipeList;
+import seedu.address.model.item.Item;
+import seedu.address.model.item.Quantity;
+import seedu.address.model.recipe.Ingredient;
+import seedu.address.model.recipe.IngredientList;
+import seedu.address.model.recipe.Recipe;
+import seedu.address.testutil.RecipeBuilder;
+
+public class CheckCraftCommandTest {
+
+    private ModelStubWithItemAndRecipeList model;
+    private ModelStubWithItemAndRecipeList expectedModel;
+
+    @BeforeEach
+    public void setUp() {
+        IngredientList ingredients = new IngredientList();
+        ingredients.add(new Ingredient(1, 3));
+        // recipe to create 2 apples from 3 bananas
+        Recipe recipe = new RecipeBuilder().withId(1).withIngredients(ingredients).withQuantity("2")
+                .withDescription("test").build();
+        RecipeList recipeList = new RecipeList();
+        recipeList.addRecipe(recipe);
+        model = new ModelStubWithItemAndRecipeList(getTypicalItemList(), recipeList);
+        expectedModel = new ModelStubWithItemAndRecipeList(getTypicalItemList(), recipeList);
+    }
+
+    /**
+     * Tests for success when item can be crafted with 1 recipe.
+     */
+    @Test
+    public void execute_craftable_success() {
+        CheckCraftCommand ccc = new CheckCraftCommand(APPLE.getName(), new Quantity("3"));
+        String expectedMessage = String.format(MESSAGE_SUCCESS_CRAFTABLE, APPLE.getName(), 3)
+                + "\ntest: Banana[6] -> Apple[4]\n";
```

Comment:
Nitpick again, but use of Magic Strings is not recommended

Reply:
Also can reuse the same strings for the integration test as the string is the exact same one.